### DISCUSSION TITLE
Easier pipeline to use and train custom verifier models?

### DISCUSSION THREAD

**Discussion by AnkushMalaker** [dscripka/openWakeWord#97] [Ideas]
I'm thinking of atleast a simple pipeline like this:
`scripts/train_custom.py --model_name 'hey_jarvis' --model-config model_config.json --output-path <model-path>` 
> launches docker container
> Program prompts to say "hey_jarvis" from all the speakers sequentially, uses VAD to crop the wav files to required length. UX can be decided.
> Program prompts for negative data, repeats steps made for positive data.
> Programs uses docker environment and available GPUs to train
> Easily add the custom verifier model in various apps like wyoming_openwakeword

**Comment by dscripka** [comment]
This type of pipeline could certainly work, yes. And given that the verifier models are extremely lightweight to run and train, complex containers and GPUs may not even be required. That is, this could easily be implemented as a web application and run even on low-power devices like Raspberry Pis.

This concept something I've explored briefly in the past, but never to this level of accessibility. Hopefully I can return to this concept in upcoming releases!
